packages/berry-cli/sources/cli.ts

```diff
@@ -67,9 +87,15 @@ async function run() {
       for (const command of plugin.commands || [])
         command(clipanion, pluginConfiguration);
 
-    clipanion.runExit(`yarn`, process.argv.slice(2), {
-      cwd: NodeFS.toPortablePath(process.cwd()),
-    });
+    // Print the list of available scripts if a `run` command is executed without the parameters
+    const noParamsPassedToRunCommand = process.argv.slice(2).length === 1 && process.argv[2] === `run`;
```

Comment:
This won't work - the command needs to be registered into our CLI framework 🙂 Basically, the idea is that you have to switch the `name` parameter from required (`<name>`) to optional (`[name]`) and then simply check whether it's undefined or not.

Reply:
👋 @arcanis , 

I have a problem with making a `name` an optional argument. I have been reading the documentation of [`clipanion`](https://github.com/arcanis/clipanion). I did the following changes in `run.ts` file:

```Typescript
.command(`run [name] [... args] [-T,--top-level]`)
```

Unfortunately I am getting the error:

```bash
Error: Expected "-", "--", "_", or [a-z] but "." found.
    at peg$buildStructuredError (/Users/mollyiv/Projects/berry/packages/berry-cli/bundles/berry.js:39091:12)
    at peg$parse (/Users/mollyiv/Projects/berry/packages/berry-cli/bundles/berry.js:40658:11)
    at Clipanion.command (/Users/mollyiv/Projects/berry/packages/berry-cli/bundles/berry.js:37332:26)
    at module.exports.exports.default (/Users/mollyiv/Projects/berry/packages/berry-cli/bundles/berry.js:121033:6)
    at run (/Users/mollyiv/Projects/berry/packages/berry-cli/bundles/berry.js:159:17)
```

It looks like there is a parsing error from `clipanion`. 

Is the command pattern syntax that I am using incorrect? 🤔

Reply:
Ah errata - clipanion doesn't currently support optional positional arguments + rest (it really should, but that'll be for another time). So what we should do is:

```
run [... args] [-T,--top-level]
```

And then on the JS side:

```js
if (args.length === 0)
  return printRunListing();

const [name, ...rest] = args;
```

Makes sense?

Reply:
It does, thank you! 🙇 Now we can have an issue with `--json` flag:

```
.command(`run [... args] [-T,--top-level] [--json]`)
```

```
$ yarn run --json                               
Error: Couldn't find a script named "--json".

Usage: yarn run [... args] [-T,--top-level] [--json]
```

🤔

Reply:
It does, thank you! 🙇 Now we can have an issue with `--json` flag:

```
.command(`run [... args] [-T,--top-level] [--json]`)
```

```
$ yarn run --json                               
Error: Couldn't find a script named "--json".

Usage: yarn run [... args] [-T,--top-level] [--json]
```

🤔

Reply:
I took a closer look at clipanion and it seems we won't be able to implement `--json` at the moment - there's an ambiguity on whether `--json` is meant to be part of the option or `args` (for example in `yarn run foo --json`) that it isn't able to solve at the moment. I'll implement backtracking later, but for now let's put `--json` aside 🙂

Reply:
Very well, let's skip `--json` for now, thanks for your investigation 🙇